Updates related to Operator 6.0.0 release (#1290)
minio/docs: source/operations/install-deploy-manage/upgrade-minio-operator.rst

- The ``.spec.s3`` field is replaced by the ``.spec.features`` field.
- The ``.spec.credsSecret`` field is replaced by the ``.spec.configuration`` field.

>   The ``.spec.configuration`` secret should hold all the environment variables for the MinIO deployment that contain sensitive information and should not show in ``.spec.env``.

Reviewer: maybe something like this?
```suggestion
  The ``.spec.configuration`` secret holds sensitive environment variables for the MinIO deployment [such as?]
To protect this information, these environment variables are not shown in ``.spec.env``.
```

Reviewer 2: It is in fact *all* environment configuration settings. So you could link to the docs page on server envvars.

Reviewer 2: _Technically_ the `.spec.env` can potentially work, but this is what we recommend.

Author: Reworked to take in these suggestions.

Author: Actually, the PR for 6.0.0 upgrades completely removed the need for this change.
